lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -851,6 +855,12 @@ sub _generate_job_setting {
     $settings{WORKER_CLASS} = join ',', sort @classes if @classes > 0;
     $settings{uc $_} = $args->{$_} for keys %$args;
 
+    # follow the rule that handles '+' in isos post
+    for (keys %settings) {
+        if (substr($_, 0, 1) eq '+') {
+            $settings{substr($_, 1)} = delete $settings{$_};
+        }
+    }
```

Comment:
We made the mistake to allow code duplication to happen in the job handling as well as medium handling where we have already similar handling. I strongly recommend you work on reducing the duplication again before adding this change here.

Reply:
Move these code to Utils.pm as a function.

Reply:
Please don't add it to `Utils.pm`, I started moving things out of there (cf. #2837). But `Script.pm`, which is really a *CloneJob* in disguise could perhaps handle settings for both cases.